2023.7.0 (#28081)
home-assistant/home-assistant.github.io: source/_posts/2023-07-05-release-20237.markdown

description: "Services can now respond with data, a new dialog for lock, sentence triggers, fewer breaking changes, Bluetooth proxies are now the fastest option, copy 'n paste in dashboards, brand new image entities, a dashboard action to open assist, climate support for Matter, and more!"
date: 2023-07-05 00:00:00
date_formatted: "July 5, 2023"
author: Franck Nijhof
author_twitter: frenck
comments: true
categories:
  - Release-Notes
  - Core
og_image: /images/blog/2023-07/social.png
---

<a href='/integrations/#version/2023.7'><img src='/images/blog/2023-07/social.png' style='border: 0;box-shadow: none;'></a>

Home Assistant Core 2023.7! 🎉

The previous release was a big one, and this one is no different!

This release brings in a change to Home Assistant, which we consider to be one
of the biggest game changers of the past years: Services can now respond with
data! 🤯

It is such a fundamental change, which will allow for many new use cases and
opens the gates for endless possibilities. I'm so excited about this one!

But that is not all; there is a lot more in store for you! Some of my favorites:
The Bluetooth proxies are now lightning fast, copy 'n paste in dashboards, a new

> sentence trigger, and I love that an broken automation no longer disappears

Reviewer: ```suggestion
sentence trigger, and I love that a broken automation no longer disappears
```

> in thin air but now shows up in the UI marked as problematic.

Reviewer: ```suggestion
into thin air but now shows up in the UI marked as problematic.
```

- [Other noteworthy changes](#other-noteworthy-changes)
- [New integrations](#new-integrations)
- [Integrations now available to set up from the UI](#integrations-now-available-to-set-up-from-the-ui)
- [Need help? Join the community!](#need-help-join-the-community)
- [Breaking changes](#breaking-changes)
- [Farewell to the following](#farewell-to-the-following)
- [All changes](#all-changes)

Don't forget to [join our release party live stream on YouTube](https://www.youtube.com/watch?v=r0DLyU-4nWo)
5 July 2023, at 12:00 PDT / 21:00 CEST!

<lite-youtube videoid="r0DLyU-4nWo" videotitle="Home Assistant 2023.7 Release Party"></lite-youtube>

## Services can now respond

Services can now respond with data! 🎉

Yes, you read that right! This advanced feature is a huge step forward for
Home Assistant and will allow for a lot of new possibilities. It is one of the
biggest fundamental changes to Home Assistant made in the past years.

Calling a service in an automation or script was a one-way communication
to control your devices and services. For example, turn on a light, or make
an announcement on your speakers.

> So, what is the use of a service responding data? Well, it could be used to

Reviewer 2: ```suggestion
So, what is the use of a service responding with data? Well, it could be used to
```

send it to ChatGPT using the conversation process, and ask it to summarize it
and use that response to send a notification to your phone.

[@allenporter] has put this all together for you in
[this Blueprint](/blueprints/blog/2023-07/notify_agent_agenda.yaml):

{% my blueprint_import badge blueprint_url="https://www.home-assistant.io/blueprints/blog/2023-07/notify_agent_agenda.yaml" %}

This is just one example, but the possibilities are endless! It is exciting
to see what other creative use cases and new services with responses will
be added in the future!

[This feature is also added to scripts!](/docs/scripts#stopping-a-script-sequence)
So, you can make your scripts respond and use that response in your automations.
This is a great way to make your scripts more dynamic and flexible.

A special thanks to [@allenporter], who has been responsible for building out
the larger part of this feature!

[@allenporter]: https://github.com/allenporter

## New entity dialog for Locks

We are continuing to improve the entity dialogs. This time, we have reworked
the entity dialog for locks. It looks great!

It looks similar to the entity dialog for the switch, but it has a few

> extra features like asking for a unlock code and has a beautiful animation

Reviewer 2: ```suggestion
extra features like asking for an unlock code and has a beautiful animation
```

when it is (un)locking.

<img class="no-shadow" src='/images/blog/2023-07/entity-dialog-lock.gif' alt='Screen recording demonstrating the new lock entity dialog.'>

## Sentence trigger

Our [Assist](/voice_control/) is getting more and more powerful! This time,
we have added a new trigger to our automation editor that allows you to
trigger an automation when a sentence is said to the default Assist conversation
agent. 🤯

<img class="no-shadow" src='/images/blog/2023-07/sentence-trigger.png' alt='Screenshot of the new sentence trigger that can be used with automations.'>

Creating an automation based on something that is said, has never been easier.
In the above screenshot, just mentioning "what a mess" will send out the living
room vacuum to start cleaning up the mess.

## Open assist action

Another one for the Year of the Voice, a brand new action for use in dashboard
cards, has been added: The Assist action!

> <img class="no-shadow" src='/images/blog/2023-07/assist-action.png' alt='Screenshot of the new sentence trigger that can be used with automations.'>

Reviewer 2: Stale alt text.

With this action, you can directly open Assist from your dashboards using,
for example, a button and even make Assist listen right away.

> <img class="no-shadow" src='/images/blog/2023-07/assist-action.gif' alt='Screenshot of the new sentence trigger that can be used with automations.'>

Reviewer 2: Stale alt text.

<br /><br />
There are quite a few cases where this action can be useful. For example, if
you have a tablet mounted on the wall, you can use this action to open Assist
and start listening immediately using a single button touch.

## Image entities

There is a brand new entity type in town: The [image entity](/integrations/image/).

The image entity is very similar to the [camera entity](/integrations/camera/),
but instead of providing a live stream, it provides a single image. For example,
a QR code that provides the Wi-Fi credentials for your guest network, a weather
map, an image of the last motion detected by your camera, or just the cute cat
picture of the day!

> <img class="no-shadow" src='/images/blog/2023-07/image-entity.png' alt='Screenshot of an new image entity more information dialog, showing a QR code in the image.'>

Reviewer 2: ```suggestion
<img class="no-shadow" src='/images/blog/2023-07/image-entity.png' alt='Screenshot of a new image entity more information dialog, showing a QR code in the image.'>
```

While the camera entity provided similar functionality, we wanted to make an
entity tailored for these specific use cases. An image entity is much
more lightweight and efficient compared to the camera entity. Also, this keeps
our camera entities focused on providing actual camera feeds.

Support for this brand new entity has been added to the [AVM Fritz!Box Tools](/integrations/fritz/)
(for guest network access), and the [MQTT](/integrations/mqtt/) and
[Template](/integrations/template/) integrations if you want to build your own.

## Energy dashboard: Self-sufficiency

The energy dashboard has a new gauge, added by [@TillFleisch]: Self-sufficiency.

This gauge tells you how self-sufficient your home is. The more you take from
the grid, the more the value will decrease.

> <img class="no-shadow" src='/images/blog/2023-07/self-sufficiency.png' alt='Screenshot showing the latest updates and tweaks made to the integrations dashboard, making it less busy.'>

Reviewer 2: Alt text doesn't match the topic.

understand the order of the cards across the different columns in your view.
To make that easier we now show a positional number on the card when
your dashboard is in edit mode.

Added bonus: Who needs drag 'n drop when you've got click 'n ...teleport?
Just kidding, of course; however, teleporting already makes things a lot easier!
Click on the positional number to enter a new position for the card.

<img class="no-shadow" src='/images/blog/2023-07/dashboard-teleport.gif' alt='Screen recording demonstrating clicking the positional number, provides a way to move a card to a different position.'>

[@karwosts]: https://github.com/karwosts

## Select the timezone to use/display

You can now select which timezone the Home Assistant interface should use
in your user profile. Useful when traveling or when you, for example,
manage an instance for a relative in a different time zone.

<img class="no-shadow" src='/images/blog/2023-07/choose-used-timezone-in-ui.png' alt='Screenshot showing a new option in the user profile, that allows you to choose if you want Home Assistant to use the server or your current timezone.'>

> You choose to use the timezone of your current location, use the timezone that

Reviewer: ```suggestion
You choose to use the timezone of your current location, or use the timezone that
```

Author: Changed this sentence based on a suggestion by bram

that your Home Assistant server is in. The Home Assistant frontend will adjust
all of the date & times to the selected timezone.

## Iteration of the integrations dashboard

In the last release, we introduced new [integrations dashboard & pages](/blog/2023/06/07/release-20236/#integrations-dashboard--pages),
which many of you really liked. Yet, there has been some feedback, and we
have been working on improving it.

The main goal was to make it feel less busy, more familiar and consistent with
all other UI you see in Home Assistant.

<img class="no-shadow" src='/images/blog/2023-07/integration-dashboard-improvements.png' alt='Screenshot showing the latest updates and tweaks made to the integrations dashboard, making it less busy.'>

The cards shown are now more consistent with the rest of Home Assistant and the
full integration name and logo is now clickable. The cog, device and service
items have been removed, and the indicators for custom integrations

> and cloud services are no longer overlap the logos.

Reviewer 2: ```suggestion
and cloud services are no longer overlapping the logos.
```

## Unavailable automations and scripts

Automations and scripts are awesome! They can add a lot of comfort to your

> home, making your home automation feel like magic. But they have a weird quirk;

Reviewer: ```suggestion
home, making your home automation feel like magic. But they have a weird quirk:
```

> If you made an error in their configuration, they would just disappear

Reviewer: ```suggestion
if you made an error in their configuration, they would just disappear
```

so you know what is going on and what to fix.

<img class="no-shadow" src='/images/blog/2023-07/broken-automation-details.png' alt='Screenshot showing the automation editor displaying a broken automation, including the error message on what is wrong.'>

## New script dialog

When you create a new automation, you’ll get a nice dialog where you can create
a new one from scratch or use one of your imported blueprints
as a starting point.

[@karwosts] has now added the same dialog to our script editor! 🎉

<img class="no-shadow" src='/images/blog/2023-07/new-script-dialog.png' alt='Screenshot of the dialog shown when creating a new script, allowing you to select blueprints as well.'>

Being able to set up scripts from Blueprints is great, as combined with the
brand new [service response](#services-call-can-now-respond) feature, these
become a lot more powerful and a lot more useful for sharing!

You can find new script Blueprints for you to use in the
[Blueprint Exchange on our community forum](https://community.home-assistant.io/tags/c/blueprints-exchange/53/script?ascending=false&order=views).

[@karwosts]: https://github.com/karwosts

## Lightning fast Bluetooth proxies

> A great way to extend the range of Bluetooth for your Home Assistant set up,

Reviewer: ```suggestion
A great way to extend the range of Bluetooth for your Home Assistant set up
```

is by using [Bluetooth proxies](https://esphome.github.io/bluetooth-proxies/).
These Bluetooth proxies are now even faster! Like... really fast! 🚀

> You can expect to see an 40-80% decrease in network traffic between

Reviewer 2: ```suggestion
You can expect to see a 40-80% decrease in network traffic between
```

Home Assistant and the proxy device. It now bundles and sends multiple
Bluetooth advertisements in bulk, instead of sending them one by one.

Another large contributor to the speed increase is that the processing of the
advertisements is now done in Home Assistant instead of on the proxy device.
A win-win; it is faster this way and contributes to making the ESP device more
stable and reliable.

For this to work, you need [ESPHome 2023.6.0](https://esphome.io) or later.

## Matter updates

Quite a few significant changes have been made to the Matter integration.

The biggest change can be found in the way subscriptions are handled. Especially
on larger (Thread) networks, this will result in a much faster and more reliable
experience. This also affects startup time, which is now blazing fast (from
minutes to seconds).

Changes to connected Matter bridges are now instantly detected, so when adding

> a new device to a bridge (like, for example, a connected Aqara M2 hub) it will

Reviewer 2: ```suggestion
a new device to a bridge (like, for example, a connected Aqara M2 hub) the device will
```

</p>

This means if you have a [Nest Thermostat](https://support.google.com/googlenest/answer/9246551?sjid=11501903403631214779-EU)
(not the E or learning ones) with the latest firmware, you can now use it
directly with Home Assistant, entirely local without clouds!

## Reducing the risk of running into breaking changes

Breaking changes, nobody loves them, but they are sometimes necessary to move
forward. We try to keep them to an absolute minimum, but sometimes they are
inevitable.

However, we do want to reduce the risk of running into them. Therefore we have
adopted [a new policy for deprecating YAML integration configuration options][adr].

We now require that a YAML configuration option is deprecated for at least 6
months before being removed. This means you won’t be running into
a hard-breaking change if you skip a release or two.

To improve this further, if possible, an automatic migration is now required
and an issue in the repairs dashboard will be raised if the deprecation applies
to your configuration.

[adr]: https://github.com/home-assistant/architecture/blob/master/adr/0021-YAML-integration-configuration-deprecation-policy.md

## Other noteworthy changes

> There are much more easter bunnies in this release; here are some of the other

Reviewer: I'm not sure if Easter-anything is appropriate here (Easter eggs?). Maybe just `improvements`?

```suggestion
There are many more improvements in this release; here are some of the other
```

Author: Lol leftover :)

[@ludeeus]: https://github.com/ludeeus
[#94468]: https://github.com/home-assistant/core/pull/94468

{% enddetails %}

{% details "MQTT" %}

If the `initial` temperature, `min_temp`, or `max_temp` is not set, the default
temperature will be converted to the set `temperature_unit` or system default
temperature unit if this is not degrees Celsius.

([@jbouwh] - [#93965]) ([documentation](/integrations/mqtt))

[@jbouwh]: https://github.com/jbouwh
[#93965]: https://github.com/home-assistant/core/pull/93965

---

The behavior of MQTT climate when `power_command_topic` is set has been changed:

- The `hvac_mode` state attribute is no longer optimistically set if
  `power_command_topic` is defined and `climate.turn_on` or `climate.turn_off`
  is called.
- When `hvac_mode` is changed, `payload_on` or `payload_off` messages will
  not be published on the `power_command_topic`.

This also means `power_command_topic` is no longer deprecated.

([@jbouwh] - [#94832]) ([documentation](/integrations/mqtt))

[@jbouwh]: https://github.com/jbouwh
[#94832]: https://github.com/home-assistant/core/pull/94832

{% enddetails %}

{% details "MusicCast" %}

> If you have any automation depending on the `source` state attribute of the

Reviewer 2: I'd start the paragraph by saying what changed. Ie, the state attribute `source` value has changed from eg. `net_radio` to `Net Radio`.

This may affect scripts or automatons setting fan speeds to specific percentages.

([@jgrieger1] - [#92371]) ([documentation](/integrations/zwave_js))

[@jgrieger1]: https://github.com/jgrieger1
[#92371]: https://github.com/home-assistant/core/pull/92371

{% enddetails %}

If you are a custom integration developer and want to learn about breaking
changes and new features available for your integration: Be sure to follow our
[developer blog][devblog]. The following are the most notable for this release:

- [Changes to entity naming](https://developers.home-assistant.io/blog/2023/06/28/entity-name-changes)
- [Service Call API changes](https://developers.home-assistant.io/blog/2023/06/14/service-calls)
- [Service call description filters](https://developers.home-assistant.io/blog/2023/06/28/service-description-filters)

[devblog]: https://developers.home-assistant.io/blog/

## Farewell to the following

The following integrations are also no longer available as of this release:

- **SenseME** has been removed. Big Ass Fans changed the protocol in a newer
  firmware in April 2022. The [Big Ass Fans integration] can be used

>   as a replacement, after updating the latest firmware.

Reviewer 2: ```suggestion
  as a replacement, after updating to the latest firmware.
```
